src/shmem.c

```diff
@@ -149,41 +147,6 @@ static int get_dev_shm_usage(char buffer[64])
 	return percentage;
 }
 
-// Verify the PID stored during shared memory initialization is the same as ours
-// (while we initialized the shared memory objects)
-static void verify_shmem_pid(void)
-{
-	// Open shared memory settings object
-	const int settingsfd = shm_open(SHARED_SETTINGS_NAME, O_RDONLY, S_IRUSR | S_IWUSR);
-	if(settingsfd == -1)
-	{
-		log_crit("verify_shmem_pid(): Failed to open shared memory object \"%s\": %s",
-			SHARED_SETTINGS_NAME, strerror(errno));
-		exit(EXIT_FAILURE);
-	}
-
-	ShmSettings shms = { 0 };
-	if(read(settingsfd, &shms, sizeof(shms)) != sizeof(shms))
-	{
-		log_crit("verify_shmem_pid(): Failed to read %zu bytes from shared memory object \"%s\": %s",
-			sizeof(shms), SHARED_SETTINGS_NAME, strerror(errno));
-		exit(EXIT_FAILURE);
-	}
-
-	close(settingsfd);
-
-	// Compare the SHM's PID to the one we had when creating the SHM objects
-	if(shms.pid == shmem_pid)
-		return;
-
-	// If we reach here, we are in serious trouble. Terminating with error
-	// code is the most sensible thing we can do at this point
-	log_crit("Shared memory is owned by a different process (PID %d)", shms.pid);
-	check_running_FTL();
-	log_crit("Exiting now!");
-	exit(EXIT_FAILURE);
-}
-
```

Comment:
This is not needed any longer. The helt-open file descriptors paired with the new exclusive lock effectively prevents other processes from being able to "steal" the shared memory objects. In fact, it doesn't even work any longer as multiple `pihole-FTL` instances running in parallel is not (theoretically) possible.